cmd/generate/config/rules/privatekeynonpem.go

```diff
@@ -0,0 +1,22 @@
+package rules
+
+import (
+	"regexp"
+
+	"github.com/zricethezav/gitleaks/v8/config"
+)
+
+func PrivateKeyNonPEMFormat() *config.Rule {
+	// define rule
+	r := config.Rule{
+		Description: "Private Key Non PEM Format",
+		RuleID:      "private-key-non-pem-format",
+		Regex:       regexp.MustCompile(`(?i)MII[BCEJ]`),
```

Comment:
@icecream-monster Thinking about this a little more, I think this regex should be generated with [`generateUniqueToken`](https://github.com/zricethezav/gitleaks/blob/d0733f96f3df2e078da4ee42fee92c9da69962bd/cmd/generate/config/rules/rule.go#L44). Also since we want to capture the entire secret, not just part of is, would it be possible to give the secret group of the regex a range?

Reply:
@zricethezav ya! I can do that with:
```

		Regex:       generateUniqueTokenRegex(`MII[BCEJ]{1}[0-9A-Z+\/\r\n]{64,1000}={0,2}`),
		Keywords:    []string{"MII"}

``` 
However I am wondering if we can also use `generateUniqueTokenRegex` for the `func PrivateKey() *config.Rule` (which is just the PEM-format of the exactly same key with `-----BEGIN` header and `PRIVATE KEY-----` tail). Just to keep the format of finding any private key consistent? 

In order to do that, is there a way for the `generateUniqueTokenRegex` function to accept `-` as the START of the regex? It's one of the special chars that it won't accept :( (e.g., accepts `_` as a starting char but not `-` or `+`)

thanks!